internal/resource/config/datastore/jdbc_data_store.go

```diff
@@ -229,19 +342,56 @@ func toStateJdbcDataStore(con context.Context, jdbcDataStore *client.JdbcDataSto
 	return toStateObjVal, allDiags
 }
 
-func readJdbcDataStoreResponse(ctx context.Context, r *client.DataStoreAggregation, state *dataStoreResourceModel, plan *dataStoreResourceModel) diag.Diagnostics {
+func toDataSourceStateJdbcDataStore(con context.Context, jdbcDataStore *client.JdbcDataStore) (types.Object, diag.Diagnostics) {
+	var allDiags, diags diag.Diagnostics
+
+	if jdbcDataStore == nil {
+		diags.AddError("Failed to read JDBC data store from PingFederate.", "The response from PingFederate was nil.")
+		return types.ObjectNull(jdbcDataStoreAttrType), diags
+	}
+
+	connectionUrlSetVal, diags := types.SetValueFrom(con, jdbcTagConfigAttrType, jdbcDataStore.ConnectionUrlTags)
+	allDiags = append(allDiags, diags...)
+
+	jdbcAttrValue := map[string]attr.Value{
```

Comment:
If you want to slightly reduce repetition, you could put most of this into a common method. Since from what I can tell it's just the password/encrypted_password that is different

Reply:
added methods to internaltypes package for this